core/src/main/java/org/bitcoinj/crypto/HDPath.java

```diff
@@ -57,6 +58,22 @@ private enum Prefix {
             this.symbol = symbol;
         }
 
+        static Optional<Prefix> of(char c) {
+            Optional<Prefix>  prefix;
+            switch (c) {
+                case 'm': prefix = Optional.of(Prefix.PRIVATE); break;
+                case 'M': prefix = Optional.of(Prefix.PUBLIC); break;
+                default: prefix = Optional.empty();
```

Comment:
I'd use three return statements here, and remove the break and the local variable.

Reply:
When `core` is upgraded to JDK 17, IntelliJ will offer two suggestions here: "Use switch expression" and "inline redundant variable." Applying them results in:

```
            return switch (c) {
                case 'm' -> Optional.of(Prefix.PRIVATE);
                case 'M' -> Optional.of(Prefix.PUBLIC);
                default -> Optional.empty();
            };
```